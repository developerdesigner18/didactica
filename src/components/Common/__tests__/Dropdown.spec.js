import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import LxpDropdown from "../Dropdown.vue";
import { BDropdown, BDropdownItem, BImg } from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-dropdown", LxpDropdown);
localVue.component("b-dropdown", BDropdown);
localVue.component("b-dropdown-item", BDropdownItem);
localVue.component("b-img", BImg);

describe("Dropdown.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should floating styles for dropdown", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({ isFloating: true });
      await waitRAF();

      const floatingDropdown = wrapper.find(".form-floating");
      expect(floatingDropdown.exists()).toBe(true);
    });

    it("should add error class when `showError` is true", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        showError: true,
        supportText: "Please select a value"
      });
      await waitRAF();

      const errorDropdown = wrapper.find(".error");
      expect(errorDropdown.exists()).toBe(true);

      const errorMessageContent = wrapper.find(".has-support-text");
      expect(errorMessageContent.text()).toEqual("Please select a value");
    });
  });

  describe("Events", () => {
    let wrapper;

    beforeEach(async () => {
      wrapper = await getWrapper();
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it("should emit `click` when dropdown item is clicked", async () => {
      let dropdownItem = wrapper.find(".dropdown-item");
      await dropdownItem.trigger("click");
      expect(wrapper.emitted("click")).toBeTruthy();
    });
  });
});

function getWrapper() {
  return mount(LxpDropdown, {
    propsData: {
      label: "Select",
      placeholder: "Placeholder text",
      dropdownArr: ["apple", "orange", "banana"]
    },
    localVue
  });
}
