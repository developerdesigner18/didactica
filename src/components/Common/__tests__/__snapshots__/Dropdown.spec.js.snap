// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown.vue render when created  1`] = `
<div
  class=""
>
  <div
    class="lx-select-dd"
  >
    <div
      class="lx-select__div lx-select__label--show"
    >
      <label
        class="lx-select__label"
      >
        Select
      </label>
       
      <div
        class="dropdown b-dropdown lx-select__dropdown btn-group"
      >
        <!---->
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="btn dropdown-toggle btn-secondary btn-sm dropdown-toggle-no-caret"
          type="button"
        >
          <!---->
           
          <div
            class="lx-select__dd--div lx-select__dd--div__label"
          >
            <p
              class="placeholder"
            >
              
              Select
            
            </p>
             
            <span
              class="ob-dd__icon"
            >
              <img
                align="right"
                alt="arrow"
                loading="lazy"
                src="https://files.lxp.academy.who.int/didactica/assets/images/learning-space/main/expand.svg"
              />
            </span>
          </div>
           
          <!---->
        </button>
        <ul
          class="dropdown-menu"
          role="menu"
          tabindex="-1"
        >
           
          <li
            class=""
            role="presentation"
          >
            <a
              class="dropdown-item"
              href="javascript:;"
              role="menuitem"
              target="_self"
            >
              <!---->
               
              <span
                class=""
              >
                apple
              </span>
               
              <!---->
            </a>
          </li>
          <li
            class=""
            role="presentation"
          >
            <a
              class="dropdown-item"
              href="javascript:;"
              role="menuitem"
              target="_self"
            >
              <!---->
               
              <span
                class=""
              >
                orange
              </span>
               
              <!---->
            </a>
          </li>
          <li
            class=""
            role="presentation"
          >
            <a
              class="dropdown-item"
              href="javascript:;"
              role="menuitem"
              target="_self"
            >
              <!---->
               
              <span
                class=""
              >
                banana
              </span>
               
              <!---->
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="lxp-select__footer"
  >
    <!---->
  </div>
</div>
`;
